helix-core/src/main/java/org/apache/helix/task/JobContext.java

```diff
@@ -168,8 +192,12 @@ public long getPartitionFinishTime(int p) {
   }
 
   public void setPartitionTarget(int p, String targetPName) {
-    Map<String, String> map = getMapField(p, true);
-    map.put(ContextProperties.TARGET.toString(), targetPName);
+    String targetForPartition = getTargetForPartition(p);
```

Comment:
nit: no need to create this variable. You can just do:

targetPName.equals(getTargetForPartition(p));

Reply:
Done.